src/net_processing.cpp

```diff
@@ -2959,58 +2959,37 @@ bool static ProcessMessage(CNode* pfrom, const std::string& strCommand, CDataStr
     }
 
     if (strCommand == NetMsgType::PONG) {
-        int64_t pingUsecEnd = nTimeReceived;
+        // We didn't send a ping, there shouldn't be a pong
+        if (pfrom->nPingNonceSent == 0) {
+            LogPrint(BCLog::NET, "pong peer=%d: unsolicited pong\n", pfrom->GetId());
+            return true;
+        }
+
         uint64_t nonce = 0;
         size_t nAvail = vRecv.in_avail();
-        bool bPingFinished = false;
-        std::string sProblem;
+        if (nAvail != sizeof(nonce)) {
```

Comment:
Is this intentional?  sizeof should return 8 (is 8 intentional here) but we might have side effects depending on the OS.
Why use sizeof here and not >0 or just nonce?